src/Aspire.Hosting/Publishing/DockerContainerRuntime.cs

```diff
@@ -9,49 +11,117 @@ namespace Aspire.Hosting.Publishing;
-            },
-            OnErrorData = error =>
-            {
-                logger.LogInformation("docker build (stderr): {Error}", error);
-            },
-            ThrowOnNonZeroReturnCode = false,
-            InheritEnv = true
-        };
+                throw new ArgumentException("OutputPath must be provided when ImageFormat is Oci.", nameof(options));
+            }
 
-        logger.LogInformation("Running Docker CLI with arguments: {ArgumentList}", spec.Arguments);
-        var (pendingProcessResult, processDisposable) = ProcessUtil.Run(spec);
+            builderName = $"{imageName.Replace('/', '-').Replace(':', '-')}-builder";
+            var createBuilderResult = await CreateBuildkitInstanceAsync(builderName, cancellationToken).ConfigureAwait(false);
 
-        await using (processDisposable)
+            if (createBuilderResult != 0)
+            {
+                logger.LogError("Failed to create buildkit instance {BuilderName} with exit code {ExitCode}.", builderName, createBuilderResult);
+                return createBuilderResult;
+            }
+        }
+
+        try
         {
-            var processResult = await pendingProcessResult
-                .WaitAsync(cancellationToken)
-                .ConfigureAwait(false);
+            var arguments = $"buildx build --file \"{dockerfilePath}\" --tag \"{imageName}\"";
 
-            if (processResult.ExitCode != 0)
+            // Use the specific builder for OCI builds
+            if (!string.IsNullOrEmpty(builderName))
             {
-                logger.LogError("Docker build for {ImageName} failed with exit code {ExitCode}.", imageName, processResult.ExitCode);
-                return processResult.ExitCode;
+                arguments += $" --builder \"{builderName}\"";
             }
 
-            logger.LogInformation("Docker build for {ImageName} succeeded.", imageName);
-            return processResult.ExitCode;
+            // Add platform support if specified
+            if (options?.TargetPlatform is not null)
+            {
+                arguments += $" --platform \"{options.TargetPlatform.Value.ToRuntimePlatformString()}\"";
+            }
+
+            // Add output format support if specified
+            if (options?.ImageFormat is not null || !string.IsNullOrEmpty(options?.OutputPath))
+            {
+                var outputType = options?.ImageFormat switch
+                {
+                    ContainerImageFormat.Oci => "type=oci",
+                    ContainerImageFormat.Docker => "type=docker",
+                    null => "type=docker",
+                    _ => throw new ArgumentOutOfRangeException(nameof(options), options.ImageFormat, "Invalid container image format")
+                };
+
+                if (!string.IsNullOrEmpty(options?.OutputPath))
+                {
+                    outputType += $",dest=\"{options.OutputPath}/{imageName}.tar\"";
```

Comment:
How does quotes inside of quotes work here?

`--output "type=oci,dest="C:\temp\foo.tar""`

Reply:
You mean like `--output type=oci,dest="C:\temp\foo.tar"`? I think that's probably worse because there is no indication to the shell that the value is on cohesive string other than the whitespace separation. We're also pretty consistently quoting outer arguments in our CLI calls.